crates/p2p/src/swarm/mod.rs

```diff
@@ -462,9 +471,25 @@ where
             error,
         } = event
         {
-            debug!(%peer, %error, %request_id, "Failed to send response");
+            debug!(%peer, %error, %request_id, "Inbound failure");
             return Ok(());
         }
+
+        if let RequestResponseEvent::OutboundFailure {
+            peer,
+            request_id,
+            error,
+        } = event
+        {
+            debug!(%peer, %error, %request_id, "Outbound failure");
+            return Ok(());
+        }
+
+        if let RequestResponseEvent::ResponseSent { peer, request_id } = event {
+            debug!(%peer, %request_id, "Response sent");
+            return Ok(());
+        }
+
         let RequestResponseEvent::Message { peer, message } = event else {
```

Comment:
Make `match` here, so this code won't check the event type multiple times. If you worry about the nesting, move the code that handles `RequestResponseEvent::Message` event to another method